pkg/alexa/l10n/l10n.go

```diff
@@ -0,0 +1,246 @@
+	TextSnippets    Snippets
+	IntentResponses IntentResponses
+}
+
+// Key defines the type of a text key
+type Key string
+
+func (k Key) String() string {
+	return string(k)
+}
+
+// Snippets is the actual representation of key -> array of texts in locale
+type Snippets map[Key][]string
+
+// Responses is the representation of a list of IntentResponses
+type IntentResponses map[Key]IntentResponse
+
+type IntentResponse struct {
+	Samples []string
+	Title   []string
+	Text    []string
+	SSML    []string
+	Slots   map[Key]Slot
+}
+
+type Slot struct {
+	Samples             []string
+	PromptElicitations  []alexa.PromptVariations
+	PromptConfirmations []alexa.PromptVariations // TODO: is this correct?
+}
+
+type Prompts map[Key][]alexa.PromptVariations
+
+// RegisterFunc defines the functions to be passed to Register
+type RegisterFunc func(cfg *Config)
+
+// Config contains the options for Locale registration
+type Config struct {
+	DefaultLocale bool
+	FallbackFor   string
+}
+
+// AsDefault sets the given Locale the default
+func AsDefault() RegisterFunc {
+	return func(cfg *Config) {
+		cfg.DefaultLocale = true
+	}
+}
+
+// TODO Obsolete:
+// AsFallbackFor registers the locale as fallback Locale for the given Locale name
+func AsFallbackFor(name string) RegisterFunc {
+	return func(cfg *Config) {
+		cfg.FallbackFor = name
+	}
+}
+
+// Register registers a new Locale in the DefaultRegistry
+func Register(locale *Locale, opts ...RegisterFunc) error {
+	return DefaultRegistry.Register(locale, opts...)
+}
+
+// Resolve returns the matching Locale from the DefaultRegistry
+func Resolve(name string) (*Locale, error) {
+	return DefaultRegistry.Resolve(name)
+}
+
+// Register registers a new locale and fails if it already exists
+func (r *Registry) Register(l *Locale, opts ...RegisterFunc) error {
+	_, ok := r.locales[l.Name]
+	if ok {
+		return fmt.Errorf("locale %s already registered", l.Name)
+	}
+
+	// run all RegisterFuncs
+	var cfg Config
+	for _, opt := range opts {
+		opt(&cfg)
+	}
+
+	// order matters, first anything that can fail, then changing data
+	if cfg.FallbackFor != "" {
+		// fallback locale must be registered
+		orig, ok := r.locales[cfg.FallbackFor]
+		if !ok {
+			return fmt.Errorf("cannot be fallback for locale %s as it is not registered", cfg.FallbackFor)
+		}
+
+		// fallback already defined
+		if orig.Fallback != nil {
+			return fmt.Errorf("fallback for locale %s is already registered", orig.Name)
+		}
+
+		// set the fallback
+		r.locales[cfg.FallbackFor].Fallback = l
+	}
+
+	// set locale as default
+	if cfg.DefaultLocale {
+		r.defaultLocale = l.Name
+	}
+
+	r.locales[l.Name] = l
+
+	return nil
+}
+
+func (r *Registry) GetDefaultLocale() string {
+	return r.defaultLocale
+}
+func (r *Registry) GetLocales() map[string]*Locale {
+	return r.locales
+}
+
+// Resolve returns the Locale matching the given name or an error
+func (r Registry) Resolve(name string) (*Locale, error) {
```

Comment:
Either all functions are `*Registry` or all are `Registry`, you need to pick one and stick with it.